### PR TITLE
add elementwise_add_grad_grad op

### PR DESCRIPTION
**Pull request by heavengate**
**Add elementwise_add_grad_grad op for backward of backward calculation**

### CODE REVIEW

File: paddle/fluid/operators/elementwise/elementwise_add_op.cc [hunk 1]
```diff
@@ -13,10 +13,73 @@ See the License for the specific language governing permissions and
...
+    op->SetType("elementwise_add_grad");
+    op->SetInput("X", Input("X"));
+    op->SetInput("Y", Input("Y"));
+    op->SetInput(framework::GradVarName("Out"), OutputGrad("Out"));
+    op->SetAttrMap(Attrs());
+    op->SetOutput(framework::GradVarName("X"), InputGrad("X"));
+    op->SetOutput(framework::GradVarName("Y"), InputGrad("Y"));
+    return op;
+  }
+};
```
Comment (qingqing01) [inline paddle/fluid/operators/elementwise/elementwise_add_op.cc]: 参考 https://github.com/PaddlePaddle/Paddle/pull/17353/files 使用 
```
REGISTER_ELEMWISE_GRAD_MAKER(elementwise_add, Add);
```

生成ElementwiseAddOpGradDescMaker，避免二次代码写错。

File: paddle/fluid/operators/elementwise/elementwise_add_op.h [hunk 2]
```diff
@@ -160,5 +160,31 @@ class ElementwiseAddGradKernel : public ElemwiseGradKernel<T> {
...
+    // ddOut = ddx + ddy
+    if (ddout) {
+      Tensor ddx_safe, ddy_safe;
+      GetDoubleGradSafeTensor<DeviceContext, T>(ctx, x, ddx, &ddx_safe);
+      GetDoubleGradSafeTensor<DeviceContext, T>(ctx, y, ddy, &ddy_safe);
+
+      ddout->mutable_data<T>(ctx.GetPlace());
+      default_elementwise_add<DeviceContext, T>(ctx, &ddx_safe, &ddy_safe,
+                                                ddout);
+    }
```
Comment (qingqing01) [inline paddle/fluid/operators/elementwise/elementwise_add_op.h:186]: Whether needs to set  zero for output `framework::GradVarName("X")` and `framework::GradVarName("Y")` ?

File: paddle/fluid/operators/elementwise/elementwise_add_op.cc (hunk 1 above)
Comment (heavengate) [reply paddle/fluid/operators/elementwise/elementwise_add_op.cc]: Done. Thanks

File: paddle/fluid/operators/elementwise/elementwise_add_op.h (hunk 2 above)
Comment (heavengate) [reply paddle/fluid/operators/elementwise/elementwise_add_op.h:186]: removed `framework::GradVarName("X")` and `framework::GradVarName("Y")` as output

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**